### PR TITLE
Frontend private room

### PR DESCRIPTION
**Pull request by jcderry5**
This creates the html page for the private room and plays a hard-coded youtube video on the private room page

### CODE REVIEW

File: src/main/java/com/google/sps/servlets/JoinRoomServlet.java [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+  private String inputtedUserTag = "user-party-link";
+  private String currentRoomId = "";
+
+  @Override
+  public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    // TODO: Create hard-coded hashmap of {Room ID : URL} Hashmap
+    // TODO: Print json-ified string to /join-room page for private room to fetch
+  }
+
+	@Override
```
Comment (KillianLeC) [inline src/main/java/com/google/sps/servlets/JoinRoomServlet.java:39]: Indentation

File: src/main/java/com/google/sps/servlets/JoinRoomServlet.java [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+  public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+		// Take in user's private room and store it to global variable
+		currentRoomId = request.getParameter(inputtedUserTag);
+		response.getWriter().println(currentRoomId);
+		// TODO: Redirect to private room Page
+		response.sendRedirect("/views/private-room.html");
```
Comment (KillianLeC) [inline src/main/java/com/google/sps/servlets/JoinRoomServlet.java:45]: indentation

File: src/main/java/com/google/sps/servlets/JoinRoomServlet.java [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+  }
+}
```
Comment (KillianLeC) [inline src/main/java/com/google/sps/servlets/JoinRoomServlet.java:47]: nit: newline

File: src/main/java/com/google/sps/servlets/JoinRoomServlet.java (hunk 2 above)
Comment (KillianLeC) [inline src/main/java/com/google/sps/servlets/JoinRoomServlet.java:45]: Create a global variable in a util file that contains this html path. Then use it here.

Useful if you ever change the path to this html file in the future.

File: src/main/java/com/google/sps/servlets/JoinRoomServlet.java [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+  public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    // TODO: Create hard-coded hashmap of {Room ID : URL} Hashmap
+    // TODO: Print json-ified string to /join-room page for private room to fetch
+  }
+
+	@Override
+  public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+		// Take in user's private room and store it to global variable
+		currentRoomId = request.getParameter(inputtedUserTag);
+		response.getWriter().println(currentRoomId);
```
Comment (KillianLeC) [inline src/main/java/com/google/sps/servlets/JoinRoomServlet.java:43]: Make sure to handle null currentRoomId, or invalid one?

File: src/main/webapp/scripts/private-room-script.js [hunk 5]
```diff
@@ -0,0 +1,62 @@
...
+firstScriptTag.parentNode.insertBefore(tag, firstScriptTag);
+
+// 3. This function creates an <iframe> (and YouTube player)
+//    after the API code downloads.
+var player;
+function onYouTubeIframeAPIReady() {
+	player = new YT.Player('player', {
+		height: '390',
+		width: '640',
+		videoId: 'M7lc1UVf-VE',
```
Comment (KillianLeC) [inline src/main/webapp/scripts/private-room-script.js:37]: please add TODO for the videoId being mutable.

File: src/main/webapp/scripts/private-room-script.js [hunk 6]
```diff
@@ -0,0 +1,62 @@
...
+
+// 4. The API will call this function when the video player is ready.
+function onPlayerReady(event) {
+	event.target.playVideo();
+}
+
+// 5. The API calls this function when the player's state changes.
+//    The function indicates that when playing a video (state=1),
+//    the player should play for six seconds and then stop.
+var done = false;
```
Comment (KillianLeC) [inline src/main/webapp/scripts/private-room-script.js:53]: move global variables to top of the file.

Also please rename to be more descriptive.

File: src/main/webapp/views/join-room.html [hunk 7]
```diff
@@ -1,8 +1,15 @@
-<!DOCTYPE html>
+  <!DOCTYPE html>
```
Comment (KillianLeC) [inline src/main/webapp/views/join-room.html:1]: indentation

File: src/main/webapp/views/join-room.html [hunk 8]
```diff
@@ -1,8 +1,15 @@
...
   <head>
     <meta charset="UTF-8">
     <title>Join Youtube Party Room</title>
+    <style>
+      body{
+          margin: auto;
+          text-align: center;
+      }
+    </style>
+
```
Comment (KillianLeC) [inline src/main/webapp/views/join-room.html:12]: remove newline

File: src/main/webapp/views/join-room.html [hunk 9]
```diff
@@ -1,8 +1,15 @@
...
 <html>
   <head>
     <meta charset="UTF-8">
     <title>Join Youtube Party Room</title>
+    <style>
+      body{
+          margin: auto;
+          text-align: center;
+      }
+    </style>
```
Comment (KillianLeC) [inline src/main/webapp/views/join-room.html:11]: create css file.

File: src/main/webapp/views/join-room.html [hunk 10]
```diff
@@ -13,4 +20,4 @@ <h2>Input your private room link Below!</h2>
       </form>
     </div>   
   </body>
-</html>
+</html>
```
Comment (KillianLeC) [inline src/main/webapp/views/join-room.html:23]: newline end of file

File: src/main/webapp/scripts/private-room-script.js (hunk 6 above)
Comment (jcderry5) [reply src/main/webapp/scripts/private-room-script.js:53]: I thought variables were supposed to be declared right before used?

File: src/main/java/com/google/sps/servlets/JoinRoomServlet.java [hunk 11]
```diff
@@ -32,15 +32,11 @@
 
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    // Take in user's private room and store it to global variable
+	currentRoomId = request.getParameter(inputtedUserTag);
+	response.getWriter().println(currentRoomId);
     // TODO: Create hard-coded hashmap of {Room ID : URL} Hashmap
     // TODO: Print json-ified string to /join-room page for private room to fetch
+	response.sendRedirect(JoinRoomUtil.PRIVATE_ROOM_PATH);
```
Comment (KillianLeC) [inline src/main/java/com/google/sps/servlets/JoinRoomServlet.java:40]: indentation

File: src/main/java/com/google/sps/servlets/JoinRoomUtil.java [hunk 12]
```diff
@@ -0,0 +1,6 @@
+package com.google.sps.servlets;
+import com.google.gson.Gson;
+
+public class JoinRoomUtil {
```
Comment (KillianLeC) [inline src/main/java/com/google/sps/servlets/JoinRoomUtil.java:4]: Make this class name more generic, like ServletUtil.

Also, add class level documentation

File: src/main/webapp/scripts/private-room-script.js [hunk 13]
```diff
@@ -0,0 +1,52 @@
...
+  console.log('Fetching private room.');
+  // TODO: fetch hard-coded private room
+}
+
+// This code loads the IFrame Player API code asynchronously.
+var scriptTag = document.createElement('script');
+
+scriptTag.src = "https://www.youtube.com/iframe_api";
+var firstScriptTag = document.getElementsByTagName('script')[0];
+firstScriptTag.parentNode.insertBefore(tag, firstScriptTag);
```
Comment (KillianLeC) [inline src/main/webapp/scripts/private-room-script.js:14]: move this to top of file. Have all global variables above any global statements (code).

File: src/main/webapp/scripts/private-room-script.js [hunk 14]
```diff
@@ -0,0 +1,52 @@
...
+
+/**
+* This function creates an <iframe> (and YouTube player)
+* after the API code downloads.
+* TODO: Make videoId mutable
+*/
+var player;
```
Comment (KillianLeC) [inline src/main/webapp/scripts/private-room-script.js:21]: top of file. Also, please rename this variable to be youtubePlayer.

File: src/main/webapp/scripts/private-room-script.js [hunk 15]
```diff
@@ -0,0 +1,52 @@
...
+
+// The API will call this function when the video player is ready.
+function onPlayerReady(event) {
+	event.target.playVideo();
+}
+
+// The API calls this function when the player's state changes.
+// The function indicates that when playing a video (state=1),
+// the player should play for six seconds and then stop.
+var videoFinished = false;
```
Comment (KillianLeC) [inline src/main/webapp/scripts/private-room-script.js:42]: isVideoFinished, move the variable at top of file.

File: src/main/java/com/google/sps/servlets/JoinRoomServlet.java [hunk 16]
```diff
@@ -32,15 +32,11 @@
...
     // TODO: Print json-ified string to /join-room page for private room to fetch
-  }
-
-  @Override
-  public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
-		// Take in user's private room and store it to global variable
-		currentRoomId = request.getParameter(inputtedUserTag);
-		response.getWriter().println(currentRoomId);
-		// TODO: Redirect to private room Page
+	response.sendRedirect(JoinRoomUtil.PRIVATE_ROOM_PATH);
```
Comment (KillianLeC) [inline src/main/java/com/google/sps/servlets/JoinRoomServlet.java:40]: ServletUtil.PRIVATE_ROOM_PATH

File: src/main/webapp/scripts/private-room-script.js (hunk 13 above)
Comment (KillianLeC) [reply src/main/webapp/scripts/private-room-script.js:14]: Not resolved, please update. If you disagree, please add to conversation here.

File: src/main/webapp/scripts/private-room-script.js (hunk 14 above)
Comment (KillianLeC) [reply src/main/webapp/scripts/private-room-script.js:21]: not resolved.

File: src/main/webapp/scripts/private-room-script.js (hunk 5 above)
Comment (KillianLeC) [reply src/main/webapp/scripts/private-room-script.js:26]: not resolved.

File: src/main/webapp/scripts/private-room-script.js (hunk 15 above)
Comment (KillianLeC) [reply src/main/webapp/scripts/private-room-script.js:42]: not resolved.

**Pull request merged**